scripts/build_benchmark_configs.sh

```diff
@@ -0,0 +1,103 @@
+# locations.
+
+set -eo pipefail
+
+SCRIPTPATH=$(dirname $(readlink -f $0))
+SDKPATH=$(realpath ${SCRIPTPATH}/../sdk)
+
+BOARD=$(realpath $1)
+BENCHMARK=$(realpath $2)
+SDK=$(realpath $3)
+BOARDPATH=$(dirname ${BOARD})
+BOARDNAME=$(basename -s .json ${BOARD})
+
+if [ ! -f "${BOARD}" ] ; then
+	echo "First argument must be the path of a board description file.  '${BOARD}' is not a valid file"
+	exit 1
+fi
+if [ ! -f "${BENCHMARK}/xmake.lua" ] ; then
+	echo "Second argument must be the path of a benchmark file.  '${BENCHMARK}' does not contain an xmake.lua file"
+	exit 1
+fi
+if [ ! -d "${SDK}" ] ; then
+	echo "Third argument must be the path of the sdk.  '${SDK}' does not exist"
+	exit 1
+fi
+if [ ! -f "${SDK}/bin/clang" ] ; then
+	echo "Third argument must be the path of the sdk.  '${SDK}/bin/clang' does not exist"
+	exit 1
+fi
+
+# Set up a temporary directory to work in.
+DIR=$(mktemp -d benchmark.XXX) || exit 1
+echo Working in ${DIR}...
+DIR=$(pwd)/${DIR}
+cd  ${DIR}
+
+
+# Create a directory for the board files
+
+mkdir boards
+cd boards
+
+
+# Our JSON files are slightly extended JSON, parse them with xmake's JSON
+# parser and spit them out as standard JSON so that jq can consume them.
+cp ${BOARD} basic.json
+cat <<EOF > xmake.lua
+target('fake')
+	on_load(function (target)
+		import("core.base.json")
+		local board = json.loadfile(path.join(os.scriptdir(), 'basic.json'))
+		json.savefile(path.join(os.scriptdir(), 'standard.json'), board)
+	end
+	)
+EOF
+xmake f > /dev/null
+rm xmake.lua basic.json
+rm -rf .xmake
+
+# Create variants of the board with different versions of the revoker and with
+# and without the stack high watermark.
+
+jq '.revoker="software" | .stack_high_water_mark=false' < standard.json > ${BOARDNAME}-software-revoker.json
+jq '.revoker="software" | .stack_high_water_mark=true' < standard.json > ${BOARDNAME}-software-revoker-shwm.json
+jq '.revoker="hardware" | .stack_high_water_mark=false' < standard.json > ${BOARDNAME}-hardware-revoker.json
+jq '.revoker="hardware" | .stack_high_water_mark=true' < standard.json > ${BOARDNAME}-hardware-revoker-shwm.json
+jq 'del(.revoker) | .stack_high_water_mark=false' < standard.json > ${BOARDNAME}-no-revoker.json
+jq 'del(.revoker) | .stack_high_water_mark=false | .defines[.defines| length]|= .+"CHERIOT_FAKE_REVOKER"' < standard.json > ${BOARDNAME}-fake-revoker.json
+rm standard.json
+
+CONFIGS="${BOARDNAME}-software-revoker ${BOARDNAME}-software-revoker-shwm ${BOARDNAME}-hardware-revoker ${BOARDNAME}-hardware-revoker-shwm ${BOARDNAME}-no-revoker ${BOARDNAME}-fake-revoker"
+
+# Patch the include directories so that relative paths become absolute relative
+# to the board file.
+for I in ${CONFIGS}; do
+	# Fix up includes if they expect to be relative to the board file path
+	jq ".driver_includes = (.driver_includes | map_values(. = if . | startswith(\"/\") then . else \"${BOARDPATH}/\" + . end))" < $I.json > $I.json.fixed
```

Comment:
I think this also needs to probe for starting with `$`, as in `${sdk}`, which will be expanded by `xmake`?
```suggestion
	jq ".driver_includes = (.driver_includes | map_values(. = if . | startswith(\"/\") or startswith(\"\$\") then . else \"${BOARDPATH}/\" + . end))" < $I.json > $I.json.fixed
```

Reply:
No, that will do the wrong thing. If include paths are sdk-relative then we need to do sdk substitution. I had an if then else thing in the jq that substituted ${SDKDIR} for the literal ${sdk} in that case. If you have a file that uses sdk that you can test with, can you try adding that back?

Reply:
You need to have it inside the map_values bit.  This line is trying to replace the array, not the elements of the array.

Reply:
Yes, that's correct.  I was misremembering why I didn't do this.  We only do ${sdk} substitution for include files and treat relative paths as relative to the board source location.  For the simulator, we also allow ${board}.  That's the thing that needs updating when we move the board file.  Ho hum.